python/ray/serve/_private/http_proxy.py

```diff
@@ -159,6 +159,25 @@ def update_routes(self, endpoints: Dict[EndpointTag, EndpointInfo]) -> None:
         self.route_info = route_info
         self.app_to_is_cross_language = app_to_is_cross_language
 
+    def match_target(self, target: str) -> Optional[str]:
+        """Return the route from the target.
```

Comment:
What is a target and how does it differ from a route?

Reply:
So `self.route_info` looks something like this
```
{
    '/': ('app1_http-deployment', 'app1'), 
    '/foo': ('app2_http-deployment', 'app2'), 
    '/bar': ('app3_grpc-deployment', 'app3'), 
    '/baz': ('app4_grpc-deployment-model-composition', 'app4')
}
```

`route` is the the keys such as `/`, `/foo`. `target` is matching with the names such as `app1_http-deployment` and `app2_http-deployment`

Reply:
I can change `target` to `target_endpoint` if that clear things a bit, but I feel either name is not the best description of what it does 😅